puffin/src/lib.rs

```diff
@@ -859,31 +1071,27 @@ macro_rules! profile_function {
 /// An optional second argument can be a string (e.g. a mesh name) to help diagnose what was slow.
 /// Example: `profile_scope!("load_mesh", mesh_name);`
 ///
-/// Overhead: around 140 ns on 2020 Intel MacBook Pro.
+/// Overhead: around 54 ns on Macbook Pro with Apple M1 Max.
 #[macro_export]
 macro_rules! profile_scope {
-    ($id:expr) => {
-        $crate::profile_scope!($id, "");
+    ($name:expr) => {
+        $crate::profile_scope!($name, "");
     };
-    ($id:expr, $data:expr) => {
+    ($name:expr, $data:expr) => {
```

Comment:
should `name` be `literal` rather than `expr` here?
if `expr` allows variable expressions here things might get confusing?

Reply:
Ok. I can see how that would be useful.
Perhaps we should add to the doccomment that we only use the first invocation of the expression?

Reply:
I added it because in our large code base it seems someplace require it to be an expr and it relaxes the constraints a bit on how names are generated and were they come from.